joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -258,12 +258,26 @@ controller_interface::return_type JointTrajectoryController::update(
           // Update PIDs
           for (auto i = 0ul; i < dof_; ++i)
           {
+            // If effort interface only, add desired effort as feed forward
+            // If velocity interface, ignore desired effort
             tmp_command_[i] = (state_desired_.velocities[i] * ff_velocity_scale_[i]) +
+                              (has_effort_command_interface_ ? state_desired_.effort[i] : 0.0) +
```

Comment:
```suggestion
            // If effort interface only, add desired effort as feed-forward
            // If velocity interface, ignore desired effort
            tmp_command_[i] = (state_desired_.velocities[i] * ff_velocity_scale_[i]) +
                              (has_velocity_command_interface_ ? 0.0 : (has_effort_command_interface_ ? state_desired_.effort[i] : 0.0)) +
```

Reply:
I admit this one is a not very readable.
It works because the whole block runs only if `use_closed_loop_pid_adapter_ == true` (line 256, just above the displayed preview).
This is set [here](https://github.com/VladimirIvan/ros2_controllers/blob/fedd208c60f7a3aff8d19bc841d1f412746edddb/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L719) if `has_velocity_command_interface_ || has_effort_command_interface_` as long as there is only one command interface. So checking for only for velocity command interface gives the correct result. It is probably an open door for all sorts of bugs though. I'll go with the suggestion.